lib/puppet/pops/types/types.rb

```diff
@@ -2799,6 +2800,12 @@ def really_instance?(o, guard = nil)
     resolved_type.really_instance?(o, guard)
   end
 
+  # @return `nil` to prevent serialization of the type_expr used when first initializing this instance
+  # @api private
+  def type_expr
+    nil
```

Comment:
The Object reader/writer assumes that all attributes that are passed to the initializer also can be accessed using an attribute reader. This particular attribute is very special since it is volatile (the expression is only used in the creation of the alias and then meaningless after the expression has been resolved).